src/shogun/base/AnyParameter.h

```diff
@@ -145,13 +149,23 @@ namespace shogun
 		}
 		explicit AnyParameter(const Any& value) : m_value(value), m_properties()
 		{
+			m_init_function = [& m_value = m_value]() { return m_value; };
```

Comment:
I think capturing `this` makes the code more self-explanatory. I'm curious, any reason to prefer this syntax? I wonder if this `[&attribute = attribute]` is documented somewhere actually, I only found it mentioned in stack overflow.

Reply:
I thought this is captured by value? I thought that capturing the member by reference would be more efficient? The `[&attribute = attribute]` isn't well documented, I agree, I just found it in someones else's code a while ago and started using it..

Reply:
are you referring to [this](https://stackoverflow.com/a/42029220)?

Reply:
That's a bit the problem also, if it is not well documented, it is a bit trickier to figure out if it is captured by reference or value ;-) Anyhow, I guess that the leading & indicates it is by reference.

Reply:
what is weird is that in my case the dangling reference error happened only when I cast the referenced value in the lambda . Which might be a problem now that I think about it?